src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -439,7 +439,6 @@ inline oop ShenandoahHeap::try_evacuate_object(oop p, Thread* thread, Shenandoah
       if (from_region->is_young()) {
         // Signal that promotion failed. Will evacuate this old object somewhere in young gen.
         report_promotion_failure(thread, size);
-        handle_promotion_failure();
```

Comment:
The only thing `handle_promotion_failure` did was set a flag on the old heuristic that would cause it to trigger an old collection. That flag is no longer used so I removed all that was setting it.